util/fmtutil/format.go

```diff
@@ -140,11 +141,23 @@ func makeTimeseries(wr *prompb.WriteRequest, labels map[string]string, m *dto.Me
 		for key, value := range labels {
 			bucketLabels[key] = value
 		}
+		var hasInf bool
 		for _, b := range m.GetHistogram().Bucket {
+			if b.GetUpperBound() == math.Inf(1) {
+				hasInf = true
+			}
 			bucketLabels[model.MetricNameLabel] = metricName + bucketStr
 			bucketLabels[model.BucketLabel] = fmt.Sprint(b.GetUpperBound())
 			toTimeseries(wr, bucketLabels, timestamp, float64(b.GetCumulativeCount()))
 		}
+
+		// Add +Inf bucket if not present
+		if !hasInf {
+			bucketLabels[model.MetricNameLabel] = metricName + bucketStr
+			bucketLabels[model.BucketLabel] = fmt.Sprint(math.Inf(1))
```

Comment:
This is slightly simpler, no need to do math.Inf

```suggestion
			bucketLabels[model.BucketLabel] = "+Inf"
```

Reply:
Fixed